server/controllers/finance/reports/shared.js

```diff
@@ -221,6 +224,16 @@ function formatFilters(qs) {
       if (filter.isPeriod) {
         const service = new PeriodService(new Date());
         filter.value = service.periods[value].translateKey;
+      } else if (filter.isDepot) {
+        db.one('SELECT text FROM depot WHERE uuid = ?', db.bid(value))
```

Comment:
This function isn't asynchronous.  This might work, sometimes, but it won't work all the time.

Typically what we do is send back the depotUuid and inventoryUuid as "displayValues" mapping.   This isn't well developed so it may not even work.

I think that to make this work properly we either need to:

1. Send the proper display value from the client (this is the half-hearted implementation we currently have)
2. Make a standard filter definition for anything that could be a uuid -> text label.  So, for example, we could have "lookupHumanReadableLabelByUuid()" which scans the document_map, the entity_map, the inventory, etc, to figure out what the label should be and then displays that.  This would be a server-side function and used in things like reporting.  We could then just change all these to `isUuid`, make this filter options `async`, and then have a standard way of looking up the labels by their uuid.

What are you thoughts?

Reply:
My thoughts:
- Although db.one() is asynchronous, I knew the `formatFilters()` was not.  Perhaps it should be!  I notice that the server-side report JS file does awaits on other things.  So it could easily do an awit for `formatFilters()`.
- I don't like sending the display text with things that have uuids.   I think we should be looking the labels up on the server side.
- Your second option seems like overkill to me...